pkg/utils/utils_test.go

```diff
@@ -0,0 +1,77 @@
+package utils_test
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	utils "github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/utils"
+)
+
+var _ = Describe("HashConfigMap", func() {
+	It("should hash the ConfigMap correctly", func() {
+		data := make(map[string]string)
+		data["key1"] = "value1"
+		data["key2"] = "value2"
+		cm := &corev1.ConfigMap{
+			Data: data,
+		}
+
+		expectedHash := "7cb7a94f45100d7dc8aadffbcd409f25"
+
+		actualHash := utils.HashConfigMap(cm)
+
+		Expect(actualHash).To(Equal(expectedHash))
+	})
+
+	It("Should not change hash for different resource versions", func() {
+		data := make(map[string]string)
+		data["key1"] = "value1"
+		data["key2"] = "value2"
+
+		cm1 := &corev1.ConfigMap{
+			Data: data,
+			ObjectMeta: metav1.ObjectMeta{
+				ResourceVersion: "68790",
+			},
+		}
+
+		cm2 := &corev1.ConfigMap{
+			Data: data,
+			ObjectMeta: metav1.ObjectMeta{
+				ResourceVersion: "69889",
+			},
+		}
+
+		hash1 := utils.HashConfigMap(cm1)
+		hash2 := utils.HashConfigMap(cm2)
+
+		Expect(hash1).To(Equal(hash2))
+	})
+
+	It("should not change hash for different key orderings", func() {
+		data1 := map[string]string{}
+		data1["key1"] = "value1"
```

Comment:
Maybe another test case to test if the data array or empty or each array element is zero.

Reply:
Can you elaborate?

Reply:
How does the hash function work if it finds 0 keys. And if it does find some keys. What if the contents are nil.

Reply:
hash of empty thing is still a hash. Can a config map contain nil values? I don't think so. The function will still work, but it just ignores all nil values so no need to handle it in a special way imo.

Reply:
Me and @bn222 talked offline. The first test case should cover hashing consistency. This comment is resolved.